### PR TITLE
refactor: Migrate Errors to TypeScript

### PR DESCRIPTION
**Pull request by bardiawire**

### CODE REVIEW

**Comment by bardiawire** [comment]
> Please remove all imports from [`globals.ts`](https://github.com/wireapp/wire-webapp/blob/dev/src/script/main/globals.ts) for the files which are now exported as modules.

@ffflorian Done in: https://github.com/wireapp/wire-webapp/pull/8508/commits/10bf8a8371b6e886328cdc2cd411cd7ea993abb8

**Comment by bardiawire** [comment]
it was an import error, fixed in https://github.com/wireapp/wire-webapp/pull/8508/commits/785a50b84d0cfe497a66d927d6f9a739d00c11cf

**Comment by bardiawire** [comment]
it was an import error, fixed in https://github.com/wireapp/wire-webapp/pull/8508/commits/785a50b84d0cfe497a66d927d6f9a739d00c11cf

File: src/script/error/BaseError.ts [hunk 1]
```diff
@@ -17,38 +17,41 @@
...
+  type: BASE_ERROR_TYPE | string;
+
+  constructor(name: string, type: BASE_ERROR_TYPE | string, message: string) {
     super();
 
-    this.name = this.constructor.name;
     this.stack = new Error().stack;
+    this.name = this.constructor.name;
+    this.message = message || BaseError.MESSAGE[type];
+    this.type = type ? type : BaseError.TYPE.UNKNOWN;
```
Comment (ffflorian) [inline src/script/error/BaseError.ts]:
```suggestion
    this.type = type || BaseError.TYPE.UNKNOWN;
```

Comment (bardiawire) [reply src/script/error/BaseError.ts]: After talking to @bennyn we decided now that this is migrated to TypeScript let's just go with the required parameters so there would be no more needing to fall back to default things. I'll add a few commits and let you know to please take a look again.

Comment (ffflorian) [reply src/script/error/BaseError.ts]: Alright!

**Comment by bennycode** [comment]
Test results look good on https://wire-webapp-qa.zinfra.io/

**Comment by bennycode** [comment]
Had to revert because of Nightly test run. Let's have a look at it on Tuesday!

**Pull request merged**